### PR TITLE
Add checksum for py-more-itertools@8.12.0 and fix python dependency

### PR DESCRIPTION
**Pull request by aandvalenzuela**
Python dependency has been fixed following [#578](https://github.com/more-itertools/more-itertools/issues/578)

### CODE REVIEW

File: var/spack/repos/builtin/packages/py-more-itertools/package.py [hunk 1]
```diff
@@ -21,7 +22,8 @@ class PyMoreItertools(PythonPackage):
     version('4.1.0',  sha256='c9ce7eccdcb901a2c75d326ea134e0886abfbea5f93e91cc95de9507c0816c44')
     version('2.2',    sha256='93e62e05c7ad3da1a233def6731e8285156701e3419a5fe279017c429ec67ce0')
 
-    depends_on('python@3.5:', when='@7.1:', type=('build', 'run'))
+    depends_on('python@3.6:', when='@8.12:', type=('build', 'run'))
+    depends_on('python@3.5:', when='@7.1:8.11', type=('build', 'run'))
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-more-itertools/package.py]:
```suggestion
    depends_on('python@3.6:', when='@8.11', type=('build', 'run'))
    depends_on('python@3.5:', when='@7.1:', type=('build', 'run'))
```
From reading the issue you linked to, my understanding is that the developers stopped testing Python 3.5 in 8.6, added a feature that broke 3.5 support in 8.11 (f-strings), then removed those in 8.12 so Python 3.5 is once again supported. So 8.11.0 is the only version that requires 3.6+.

**Review by adamjstewart — CHANGES_REQUESTED** [review]

File: var/spack/repos/builtin/packages/py-more-itertools/package.py (hunk 1 above)
Comment (aandvalenzuela) [reply var/spack/repos/builtin/packages/py-more-itertools/package.py]: Ups, sorry! I misread the issue

File: var/spack/repos/builtin/packages/py-more-itertools/package.py [hunk 2]
```diff
@@ -21,6 +22,7 @@ class PyMoreItertools(PythonPackage):
     version('4.1.0',  sha256='c9ce7eccdcb901a2c75d326ea134e0886abfbea5f93e91cc95de9507c0816c44')
     version('2.2',    sha256='93e62e05c7ad3da1a233def6731e8285156701e3419a5fe279017c429ec67ce0')
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-more-itertools/package.py:24]: Can you also add https://github.com/more-itertools/more-itertools/issues/578 as a comment above this line so we rememeber why the version requirement disagrees with the docs?

Comment (aandvalenzuela) [reply var/spack/repos/builtin/packages/py-more-itertools/package.py:24]: Sure!

**Review by adamjstewart — APPROVED** [review]

**Pull request merged**